source/extensions/omni.isaac.lab/omni/isaac/lab/assets/deformable_object/deformable_object.py

```diff
@@ -0,0 +1,367 @@
+# Copyright (c) 2022-2024, The Isaac Lab Project Developers.
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import annotations
+
+import torch
+from collections.abc import Sequence
+from typing import TYPE_CHECKING
+
+import carb
+import omni.physics.tensors.impl.api as physx
+from pxr import PhysxSchema, UsdShade
+
+import omni.isaac.lab.sim as sim_utils
+from omni.isaac.lab.markers import VisualizationMarkers
+
+from ..asset_base import AssetBase
+from .deformable_object_data import DeformableObjectData
+
+if TYPE_CHECKING:
+    from .deformable_object_cfg import DeformableObjectCfg
+
+
+class DeformableObject(AssetBase):
+    """A deformable object asset class.
+
+    Deformable objects are assets that can be deformed in the simulation. They are typically used for
+    soft bodies, such as stuffed animals and food items.
+
+    Unlike rigid object assets, deformable objects have a more complex structure and require additional
+    handling for simulation. The simulation of deformable objects follows a finite element approach, where
+    the object is discretized into a mesh of nodes and elements. The nodes are connected by elements, which
+    define the material properties of the object. The nodes can be moved and deformed, and the elements
+    respond to these changes.
+
+    The state of a deformable object comprises of its nodal positions and velocities, and not the object's root
+    position and orientation. The nodal positions and velocities are in the simulation frame.
+
+    Soft bodies can be `partially kinematic`_, where some nodes are driven by kinematic targets, and the rest are
+    simulated. The kinematic targets are the desired positions of the nodes, and the simulation drives the nodes
+    towards these targets. This is useful for partial control of the object, such as moving a stuffed animal's
+    head while the rest of the body is simulated.
+
+    .. _partially kinematic: https://nvidia-omniverse.github.io/PhysX/physx/5.4.1/docs/SoftBodies.html#kinematic-soft-bodies
+    """
+
+    cfg: DeformableObjectCfg
+    """Configuration instance for the deformable object."""
+
+    def __init__(self, cfg: DeformableObjectCfg):
+        """Initialize the deformable object.
+
+        Args:
+            cfg: A configuration instance.
+        """
+        super().__init__(cfg)
+
+    """
+    Properties
+    """
+
+    @property
+    def data(self) -> DeformableObjectData:
+        return self._data
+
+    @property
+    def num_instances(self) -> int:
+        return self.root_physx_view.count
+
+    @property
+    def num_bodies(self) -> int:
```

Comment:
does this property still make sense for deformables? or more to keep align with the rigid object class?

Reply:
Mostly to keep it similar to rigid bodies but probably doesn't make sense there as well 😅 . Mostly it's there for some backwards compatibility.